hoomd/mpcd/pytest/test_collide.py

```diff
@@ -130,3 +131,466 @@ def test_run(self, small_snap, simulation_factory, cls, init_args):
+            dt=0, integrate_rotational_dof=True, rigid=rigid
+        )
+        sim.operations.integrator.cell_list.shift = False
+        sim.operations.integrator.collision_method = cls(
+            period=1,
+            embedded_particles=hoomd.filter.Rigid(flags=("constituent",)),
+            **init_args,
+        )
+
+        # run simulation
+        sim.run(1)
+        new_snap = sim.state.get_snapshot()
+        if new_snap.communicator.rank == 0:
+            assert np.array_equal(properties_rigid["mass"], new_snap.particles.mass)
+            central_flag = new_snap.particles.typeid == new_snap.particles.types.index(
+                "A"
+            )
+            constit_flag = new_snap.particles.typeid == new_snap.particles.types.index(
+                "B"
+            )
+            new_velo_central = new_snap.particles.velocity[central_flag]
+            new_velo_constituent = new_snap.particles.velocity[constit_flag]
+            new_velo_mpcd = new_snap.mpcd.velocity
+            # solve for expected central particle velocity based on linear momentum
+            initial_momentum_mpcd = np.sum(velo_mpcd, axis=0)
+            initial_momentum_md = np.array(velo_rigid) * total_mass
+            initial_momentum = initial_momentum_md + initial_momentum_mpcd
+
+            final_momentum_mpcd = np.sum(new_velo_mpcd * new_snap.mpcd.mass, axis=0)
+            final_linmom_md = initial_momentum - final_momentum_mpcd
+            final_velocity_md = final_linmom_md / total_mass
+            assert np.allclose(final_velocity_md, new_velo_central)
+
+            # solve for expected angular momentum change based on solvent
+            # multiply expected change by 2 to get quaternion
+            change_momentum_mpcd = (new_velo_mpcd - velo_mpcd) * new_snap.mpcd.mass
+            expected_change_angmom_md = np.zeros(4)
+            expected_change_angmom_md[1:] = (
+                np.sum(
+                    np.cross(def_rigid["positions"], -1 * change_momentum_mpcd), axis=0
+                )
+                * 2
+            )
+            change_angmom_md = new_snap.particles.angmom[0] - angmom_rigid
+            assert np.allclose(expected_change_angmom_md, change_angmom_md)
+
+            # check the constituent velocities match the central particle
+            # since orientation is stuck at [1, 0, 0, 0], angular velocity
+            # is 0.5 * real part of angmom / inertia.
+            new_angmom = new_snap.particles.angmom[0]
+            omega = [
+                0 if i == 0 else 0.5 * a / i
+                for i, a in zip(properties_rigid["inertia"], new_angmom[1:])
+            ]
+            expected_tangential_velocity = np.cross(omega, def_rigid["positions"])
+            expected_velocity = np.add(expected_tangential_velocity, new_velo_central)
+            assert np.allclose(new_velo_constituent, expected_velocity)
+
+    def test_rigid_collide_free_nonparticipatory(
```

Comment:
This and the next test look really similarly. Is there a way to combine them into one using a parametrization on whether the free particles participate or not?